recipes-devtools/opkg-utils/files/0001-add-metadata-parsing-functionality-to-Package-class.patch

```diff
@@ -0,0 +1,66 @@
+From 234a35bfdb3b61184a61dff1c480b3d2d8debf5d Mon Sep 17 00:00:00 2001
+From: pratheekshasn <pratheeksha.s.n@ni.com>
+Date: Mon, 9 Dec 2024 13:08:04 +0530
+Subject: [PATCH] add metadata parsing functionality to Package class
+
+Upstream-Status: Inappropriate [nilrt-specific]
+
+Signed-off-by: pratheekshasn <pratheeksha.s.n@ni.com>
+---
+ opkg.py | 22 ++++++++++++++++++++++
+ 1 file changed, 22 insertions(+)
+
+diff --git a/opkg.py b/opkg.py
+index e5a65dd..c181421 100644
+--- a/opkg.py
++++ b/opkg.py
+@@ -182,6 +182,8 @@ class Package(object):
+         self.tags = None
+         self.fn = fn
+         self.license = None
++        self.user_visible = None
++        self.display_name = None
+ 
+         self.user_defined_fields = collections.OrderedDict()
+         if fn:
+@@ -291,6 +293,10 @@ class Package(object):
+                     self.sha256 = value
+                 elif name_lowercase in self.__dict__:
+                     self.__dict__[name_lowercase] = value
++                elif name_lowercase == 'uservisible':
++                    self.user_visible = value
++                elif name_lowercase == 'displayname':
++                    self.display_name = value
+                 elif all_fields:
+                     self.user_defined_fields[name] = value
+                 else:
+@@ -392,6 +398,20 @@ class Package(object):
+ 
+     def get_license(self, license):
+         return self.license
++    
++    def get_user_visible(self):
++        print("get_user_visible() UserVisible %s" % self.user_visible)
++        return self.user_visible
++    
++    def set_user_visible(self, user_visible):
++        print("set_user_visible() UserVisible %s" % user_visible)
++        self.user_visible = user_visible
++    
++    def get_display_name(self):
++        return self.display_name
++    
++    def set_display_name(self, display_name):
++        self.display_name = display_name
```

Comment:
I don't think we need to continue the original file author's mistake of making explicit accessor methods for all the class attributes. You can just remove these and use the class attribute directly.

Reply:
I have removed this now.